therapi-runtime-javadoc/src/main/java/com/github/therapi/runtimejavadoc/ClassJavadoc.java

```diff
@@ -16,52 +16,125 @@
+        }
+        this.enumConstants = Collections.unmodifiableMap(enumMap);
+
+        LinkedHashMap<MethodJavadocKey, MethodJavadoc> methodsMap = new LinkedHashMap<>();
+        if (methods != null) {
+            methods.forEach(methodJavadoc -> methodsMap.put(methodJavadoc.toMethodJavadocKey(), methodJavadoc));
+        }
+        this.methods = Collections.unmodifiableMap(methodsMap);
+
+        LinkedHashMap<MethodJavadocKey, MethodJavadoc> constructorsMap = new LinkedHashMap<>();
+        if (constructors != null) {
+            constructors.forEach(
+                    methodJavadoc -> constructorsMap.put(methodJavadoc.toMethodJavadocKey(), methodJavadoc));
+        }
+        this.constructors = Collections.unmodifiableMap(constructorsMap);
+
+        LinkedHashMap<String, ParamJavadoc> recordsMap = new LinkedHashMap<>();
+        if (recordComponents != null) {
+            recordComponents.forEach(paramJavadoc -> recordsMap.put(paramJavadoc.getName(), paramJavadoc));
+        }
+        this.recordComponents = Collections.unmodifiableMap(recordsMap);
+    }
+
+    private ClassJavadoc(String name, Comment comment, Map<String, FieldJavadoc> fields,
+                         Map<String, FieldJavadoc> enumConstants, Map<MethodJavadocKey, MethodJavadoc> methods,
+                         Map<MethodJavadocKey, MethodJavadoc> constructors, List<OtherJavadoc> other,
+                         List<SeeAlsoJavadoc> seeAlso, Map<String, ParamJavadoc> recordComponents) {
         super(name, comment, seeAlso, other);
-        this.fields = unmodifiableDefensiveCopy(fields);
-        this.enumConstants = unmodifiableDefensiveCopy(enumConstants);
-        this.methods = unmodifiableDefensiveCopy(methods);
-        this.constructors = unmodifiableDefensiveCopy(constructors);
-        this.recordComponents = unmodifiableDefensiveCopy(recordComponents);
+        this.fields = Collections.unmodifiableMap(fields);
+        this.enumConstants = Collections.unmodifiableMap(enumConstants);
+        this.methods = Collections.unmodifiableMap(methods);
+        this.constructors = Collections.unmodifiableMap(constructors);
+        this.recordComponents = Collections.unmodifiableMap(recordComponents);
     }
 
     public static ClassJavadoc createEmpty(String qualifiedClassName) {
-        return new ClassJavadoc(qualifiedClassName, null, null, null, null, null, null, null, null) {
+        return new ClassJavadoc(qualifiedClassName, null, (List<FieldJavadoc>) null, null, null, null, null, null,
+                                null) {
             @Override
             public boolean isEmpty() {
                 return true;
             }
         };
     }
 
+    ClassJavadoc createEnhancedClassJavadoc(Class<?> clazz) {
+        if (!getName().equals(clazz.getCanonicalName())) {
+            throw new IllegalArgumentException(
+                    String.format("Class `%s` does not match class doc for `%s`", clazz, getName()));
```

Comment:
STYLE NIT: Please replace all calls to `String.format` with plain old string concatenation.

Reply:
Done